services/src/main/java/org/keycloak/authentication/authenticators/browser/AbstractUsernameFormAuthenticator.java

```diff
@@ -101,7 +101,7 @@ protected void dummyHash(AuthenticationFlowContext context) {
     }
 
     public void testInvalidUser(AuthenticationFlowContext context, UserModel user) {
-        if (user == null) {
+        if (user == null || user.getServiceAccountClientLink() != null) {
```

Comment:
Is it really necessary to add this check here? Doesn't the AuthenticationProcessor prevent the login request from reaching this point?